share/spack/setup-env.sh

```diff
@@ -189,7 +189,23 @@ _sp_prefix=$(cd "$(dirname $(dirname $_sp_share_dir))" && pwd)
 _spack_pathadd PATH       "${_sp_prefix%/}/bin"
 
 _sp_sys_type=$(spack-python -c 'print(spack.architecture.sys_type())')
-_sp_dotkit_root=$(spack-python -c "print(spack.util.path.canonicalize_path(spack.config.get_config('config').get('module_roots', {}).get('dotkit')))")
-_sp_tcl_root=$(spack-python -c "print(spack.util.path.canonicalize_path(spack.config.get_config('config').get('module_roots', {}).get('tcl')))")
+
+_sp_use_custom_moduleroots="$(grep module_roots "${_sp_prefix}/etc/spack/config.yaml" > /dev/null 2>&1)"
```

Comment:
There are other levels of configuration files you should check. A `config.yaml` could also be present in `~/.spack/`, `~/.spack/<platform>/`, `etc/spack/defaults/`, and `etc/spack/<platform>`. @citibeth plans on adding project level configuration files as well. @tgamblin am I missing any scopes?